silk-rules/src/main/scala/org/silkframework/rule/input/PathInput.scala

```diff
@@ -60,7 +60,7 @@ case class PathInput(id: Identifier = Operator.generateId, path: Path) extends I
     } else {
       var index = cachedPathIndex
       if(index < 0 || index >= entity.schema.typedPaths.size || entity.schema.typedPaths(index) != path) {
```

Comment:
The last part of this condition `entity.schema.typedPaths(index) != path` is always false, because it compares TypedPath to Path.

Reply:
added .asUntypedPath